### PR TITLE
mgr/balancer: restrict automatic balancing to specific weekdays

### PR DESCRIPTION
**Pull request by xiexingguo**
We have implemented similar semantics for the scrub behaviour.
Let's try to keep pace with that.

Signed-off-by: xie xingguo <xie.xingguo@zte.com.cn>

- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

**Comment by xiexingguo** [comment]
retest this please

**Comment by xiexingguo** [comment]
retest this please

**Comment by xiexingguo** [comment]
http://pulpito.ceph.com/xxg-2019-02-18_02:16:21-rados-wip-balancer-week-distro-basic-smithi/

**Comment by votdev** [comment]
Sorry for being too late, but is it possible to add min/max fields for ``begin_weekday`` and ``end_weekday``? The Dashboard will support the management of such module options soon and it would be fine if the UI can validate/limit the user input based this information.

**Comment by xiexingguo** [comment]
@voidbag https://github.com/ceph/ceph/pull/26505

File: src/pybind/mgr/balancer/module.py [hunk 1]
```diff
@@ -221,6 +221,22 @@ class Module(MgrModule):
             'long_desc': 'This is a time of day in the format HHMM.',
             'runtime': True,
         },
+        {
+            'name': 'begin_weekday',
+            'type': 'uint',
+            'default': 0,
+            'desc': 'Restrict automatic balancing to this day of the week or later',
+            'long_desc': '0 or 7 = Sunday, 1 = Monday, etc.',
```
Comment (usefulalgorithm) [inline src/pybind/mgr/balancer/module.py]: Hi @xiexingguo, a small bug here: in Python when `time.localtime().tm_wday == 0` it's actually Monday.

Comment (xiexingguo) [reply src/pybind/mgr/balancer/module.py]: See https://github.com/ceph/ceph/pull/26505/commits/b9da0925490d4e9e964249c54e7775d9c76b1719

Comment (usefulalgorithm) [reply src/pybind/mgr/balancer/module.py]: Ah... totally missed it. Thanks and sorry for the noise!

**Pull request merged**